builtin/providers/aws/resource_aws_instance.go

```diff
@@ -1049,7 +1060,24 @@ func buildAwsInstanceOpts(
 		}
 	}
 
-	if hasSubnet && associatePublicIPAddress {
+	private_ips := d.Get("private_ips").(*schema.Set).List()
+	if len(private_ips) != 0 {
```

Comment:
I think we'll need to refactor the whole conditional block to keep handling EC2 Classic vs VPC correctly.
I'm still ok with deprecating `private_ip` in favour of `private_ips`, but we will need to add some checks for EC2 Classic && `len(private_ips) > 1` cases. I think we should error out as early as we can in such cases.

See https://github.com/hashicorp/terraform/pull/7568 which may help in making decisions. We need to be prepared for scenarios where we don't have the list of supported platforms though - I'd suggest we 1st try assigning multiple IPs and error out with an informative error message (e.g. `("Assigning %d IPs failed: %s (hint: EC2 Classic accounts don't support multiple private IPs for EC2 instances)", len(ips), err)`).

I'm aware that it may be difficult to test on EC2 Classic and many/most of us run on VPC-only accounts, but we can at least stick to expectations described in AWS docs: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-vpc.html#differences-ec2-classic-vpc

I recommend carefully reading specifically the following two sections:

 - Multiple private IP addresses
 - Accessing the Internet